writeengine/server/we_dmlcommandproc.cpp

```diff
@@ -1062,6 +1146,31 @@ uint8_t WE_DMLCommandProc::processBatchInsert(messageqcpp::ByteStream& bs, std::
 
         ++column_iterator;
       }
+
+      // MCOL-5021 Valid AUX column OID for a table is > 3000
+      // Tables that were created before this feature was added will have
+      // tableAUXColOid = 0
+      if (tableAUXColOid > 3000)
+      {
+        CalpontSystemCatalog::ColType colType;
+        colType.compressionType = execplan::AUX_COL_COMPRESSION_TYPE;
+        colType.colWidth = execplan::AUX_COL_WIDTH;
+        colType.colDataType = execplan::AUX_COL_DATATYPE;
+        WriteEngine::ColStruct colStruct;
+        WriteEngine::DctnryStruct dctnryStruct;
+        colStruct.dataOid = tableAUXColOid;
+        colStruct.tokenFlag = false;
+        colStruct.fCompressionType = colType.compressionType;
+        colStruct.colWidth = colType.colWidth;
+        colStruct.colDataType = colType.colDataType;
+        dctnryStruct.dctnryOid = 0;
+        dctnryStruct.columnOid = colStruct.dataOid;
+        dctnryStruct.fCompressionType = colType.compressionType;
+        dctnryStruct.colWidth = colType.colWidth;
+        colStructs.push_back(colStruct);
```

Comment:
I would also replace all these with 4 lines, 3 of which would be vector::emplace_back() calls + CalpontSystemCatalog::ColType initialization list ctor call.